### DISCUSSION TITLE
FE 앞으로 방향성 논의

### DISCUSSION THREAD

**Discussion by bigsaigon333** [woowacourse-teams/2021-gpu-is-mine#356] [General]
## 09-02 Thu 회의 내용 요약
1. README 작성: FE 아키텍쳐 정리, 기술 블로그 정리
2. 디렉토리 구조 고도화
    1. next.js 디렉토리 구조 참고. 우아한테크캠프 코드 참고해도 좋을듯
    2. components에서 불필요한 것들 빼내기: route가 컴포넌트에 들어있다? GpuIcon은 좀 구체화된것 같은데?
    3. domains 디렉토리 명 변경: Service 계층 같은데 hook도 보이고 스토리북도 있다..., Header 등은 components로 가야하지 않나?
3. 모바일 대응: 반응형
4. 상세조회 페이지 디자인 시안부터 다시 대시보드처럼
    1. https://pro.ant.design/docs/layout/
    2. https://material-ui.com/store/items/devias-kit-pro/
5. Storybook 강화 - Chromatic 빌드
5. Code Splitting, Lazy Loading

**Comment by 2SOOY** [comment]
성능 최적화 `1순위`
  - 번들 크기 줄이기(Code Splitting, Lazy Loading)
  - 캐싱(CloudFront, S3)
  - api 요청수 줄이기

**Reply by bigsaigon333** [reply]
우선순위 높음

**Reply by bigsaigon333** [reply]
Pair Programming

**Comment by bigsaigon333** [comment]
문서 정리
- FE Architecture 정리 -> README 반영
   - CI/CD
   - 코드 레벨(모듈간 의존성??!!)
- 기술 블로그
  - CI/CD
  - Storybook 배포
  - test
  - yarn berry migration
  - eslint, ...

**Reply by bigsaigon333** [reply]
우선순위 중간

**Comment by bigsaigon333** [comment]
상태관리툴 도입 - RTK Migration

**Reply by 2SOOY** [reply]
+ 프로젝트를 이어받는 측면에서 새롭게 마주한 사람이 이해하기 어려울 것 같음
+ 커스텀 훅, 라이브러리 자체는 이것을 해석하는데 비용이 소모됨

**Reply by 2SOOY** [reply]
Redux로 처리할 데이터 생각해보기

1. 유저 정보
2. Gpu 서버 목록
   - Job 등록 시 매번 GPU 서버를 불러옴 (GPU 서버는 변경이 잦지 않은데, 매번 요청을 보낼 필요가 없음)

**Reply by bigsaigon333** [reply]
우선순위 낮음

Pair Programming

**Comment by bigsaigon333** [comment]
디자인 `2순위`
~- 모바일 대응: 반응형~
- 상세조회 페이지 디자인 시안부터 다시 대시보드처럼
    https://pro.ant.design/docs/layout/
    https://material-ui.com/store/items/devias-kit-pro/
- Storybook 강화 - Chromatic 빌드

**Reply by bigsaigon333** [reply]
모바일 대응 우선순위 낮음

**Comment by bigsaigon333** [comment]
## 현황 이슈 리스트업

### 공통
- Favicon 만들기
- 페이지마다 타이틀 변경( https://github.com/nfl/react-helmet 활용도 괜찮을듯)
- Alert, Confirm tab 위치 변경(웹 접근성 강화)
- Error Handling

### 로그인 화면
- #353 
- 공지사항/사용법 섹션을 추가한다: 좌측에 로그인 우측에 공지사항/사용법. 상세는 링크달기
- 자동 로그아웃 기능 구현

### GPU 서버 관리 조회 페이지, Job 조회 페이지
- Table 화 하기: 더 많은 정보를 노출시킨다. 상세 버튼, 취소 버튼 Table 기능으로 통합 고려한다
- 뷰포트가 절반 넘어가면 왼쪽 헤더가 위로 올라간다
- 라이브러리 도입도 고민해본다
- Pagination 을 적용한다
- Table UI랑 비즈니스 로직 분리
- 참고 화면
![ddd](https://user-images.githubusercontent.com/31029000/131966282-5d7b2f3f-023b-4304-b640-750d944b0838.png)
- https://material-ui.com/store/collections/free-react-dashboard/
- https://javascript.plainenglish.io/best-free-react-dashboards-resources-for-every-react-developer-90b08c1f9f21

### GPU 서버 관리 등록 페이지
pass

### Job 관리 등록 페이지
- GPU 서버 선택시 서버가 뒤늦게 떠서 가운데 정렬 되어 있는 페이지가 위로 layout shift가 크게 일어난다 
  -> GPU 서버 선택란 고정 height 주기
![Screenshot_2021-09-03_16-58-40](https://user-images.githubusercontent.com/31029000/131971516-ab0edbe2-e39d-4005-8eb2-0601dfe5c4b5.png)
- Docker Hub Url Input에 Placeholder 추가하기(https://hub.docker.com/) 

### GPU 서버 상세조회 페이지
- GPU 서버 상세정보, 현재 실행중인 Job 상세정보 UI 변경 
  - 뒷배경 깔아서 카드 느낌은 주자 (최소한)
  - 요약 정보 -> 현재 실행중인 Job 정보 등으로 수정
  - 요약 정보의 실행시간에 단위 붙이기(시간)
- Subheader에서 serverId가 표시되는 걸 server이름으로 바꾸자 
  - GPU 서버 관리 > 조회 > **1** -> GPU 서버 관리 > 조회 > **serverA**
- Job 목록: Job이 없을 때 table 어떻게 보여줄지 고민 (table을 아예 없애거나 페이지네이션 버튼을 없애거나 등)


### Job 상세조회 페이지
- UI 변경: GPU 서버 상세조회 페이지와 동일
- Subheader에서 JobId를 Job이름으로 바꾸기: GPU 서버 상세조회 페이지와 동일
- 예상 실행시간 단위 붙이기: (시간)
- Log 콘솔 성능 최적화: 몇천개가 넘는 li 요소. GM 살아나고 나서 검토하자
- 10초마다 API call 호출시의 에러 핸들링(500이 오면 더 안보내야하는거 아닌가?)
- 10초마다 API Call 호출시 그래프의 zoom이 풀리는 버그가 있음

**Comment by bigsaigon333** [comment]
### 디렉토리 구조 고도화
- next.js 디렉토리 구조 참고. 우아한테크캠프 코드 참고해도 좋을듯
- components에서 불필요한 것들 빼내기: route가 컴포넌트에 들어있다? GpuIcon은 좀 구체화된것 같은데?
- domains 디렉토리 명 변경: Service 계층 같은데 hook도 보이고 스토리북도 있다..., Header 등은 components로 가야하지 않나?

**Reply by bigsaigon333** [reply]
우선순위 낮음

**Comment by bigsaigon333** [comment]
#44
